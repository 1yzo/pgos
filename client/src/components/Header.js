import React from 'react';
import moment from 'moment';
import '../styles/header.css';
import CreateOrderForm from './CreateOrderForm';

class Header extends React.Component {
    state = { showModal: false };
    
    openModal = () => {
        this.setState(() => ({ showModal: true }));
    };

    closeModal = () => {
        this.setState(() => ({ showModal: false }));
    };

    render() {
        return (
            <div className="header">
                <div className="header__title">BLUE BOTTLE COFFEE</div>
                <div className="header__container">
                    <div className="header__date">
                        <div className="header__date__month">{moment().format('MMM').toUpperCase()}</div>
                        <div className="header__date__day">{moment().format('DD')}</div>
                    </div>
                    <div className="header__subtitle">Perfectly Ground Work Orders</div>
                    <button className="header__button" onClick={this.openModal}>CREATE ORDER</button>
                    </div>
                <CreateOrderForm showModal={this.state.showModal} closeModal={this.closeModal}/>
            </div>
        );
    }
}

export default Header;