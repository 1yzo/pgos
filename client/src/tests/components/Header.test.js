import React from 'react';
import { shallow } from 'enzyme';
import Header  from '../../components/Header';

test('should render Header correctly', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
})

test('should open modal on button click', () => {
    const wrapper = shallow(<Header />);
    wrapper.find('.header__button').simulate('click');
    expect(wrapper.state('showModal')).toBe(true);
});