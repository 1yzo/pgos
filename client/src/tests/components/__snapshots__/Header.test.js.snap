// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="header__title"
>
          BLUE BOTTLE COFFEE
</div>,
        <div
          className="header__container"
>
          <div
                    className="header__date"
          >
                    <div
                              className="header__date__month"
                    >
                              DEC
                    </div>
                    <div
                              className="header__date__day"
                    >
                              31
                    </div>
          </div>
          <div
                    className="header__subtitle"
          >
                    Perfectly Ground Work Orders
          </div>
          <button
                    className="header__button"
                    onClick={[Function]}
          >
                    CREATE ORDER
          </button>
</div>,
        <Connect(CreateOrderForm)
          closeModal={[Function]}
          showModal={false}
/>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "BLUE BOTTLE COFFEE",
          "className": "header__title",
        },
        "ref": null,
        "rendered": "BLUE BOTTLE COFFEE",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="header__date"
>
              <div
                            className="header__date__month"
              >
                            DEC
              </div>
              <div
                            className="header__date__day"
              >
                            31
              </div>
</div>,
            <div
              className="header__subtitle"
>
              Perfectly Ground Work Orders
</div>,
            <button
              className="header__button"
              onClick={[Function]}
>
              CREATE ORDER
</button>,
          ],
          "className": "header__container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="header__date__month"
>
                  DEC
</div>,
                <div
                  className="header__date__day"
>
                  31
</div>,
              ],
              "className": "header__date",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "DEC",
                  "className": "header__date__month",
                },
                "ref": null,
                "rendered": "DEC",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "31",
                  "className": "header__date__day",
                },
                "ref": null,
                "rendered": "31",
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Perfectly Ground Work Orders",
              "className": "header__subtitle",
            },
            "ref": null,
            "rendered": "Perfectly Ground Work Orders",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "CREATE ORDER",
              "className": "header__button",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "CREATE ORDER",
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "closeModal": [Function],
          "showModal": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="header__title"
>
            BLUE BOTTLE COFFEE
</div>,
          <div
            className="header__container"
>
            <div
                        className="header__date"
            >
                        <div
                                    className="header__date__month"
                        >
                                    DEC
                        </div>
                        <div
                                    className="header__date__day"
                        >
                                    31
                        </div>
            </div>
            <div
                        className="header__subtitle"
            >
                        Perfectly Ground Work Orders
            </div>
            <button
                        className="header__button"
                        onClick={[Function]}
            >
                        CREATE ORDER
            </button>
</div>,
          <Connect(CreateOrderForm)
            closeModal={[Function]}
            showModal={false}
/>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "BLUE BOTTLE COFFEE",
            "className": "header__title",
          },
          "ref": null,
          "rendered": "BLUE BOTTLE COFFEE",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="header__date"
>
                <div
                                className="header__date__month"
                >
                                DEC
                </div>
                <div
                                className="header__date__day"
                >
                                31
                </div>
</div>,
              <div
                className="header__subtitle"
>
                Perfectly Ground Work Orders
</div>,
              <button
                className="header__button"
                onClick={[Function]}
>
                CREATE ORDER
</button>,
            ],
            "className": "header__container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="header__date__month"
>
                    DEC
</div>,
                  <div
                    className="header__date__day"
>
                    31
</div>,
                ],
                "className": "header__date",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "DEC",
                    "className": "header__date__month",
                  },
                  "ref": null,
                  "rendered": "DEC",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "31",
                    "className": "header__date__day",
                  },
                  "ref": null,
                  "rendered": "31",
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Perfectly Ground Work Orders",
                "className": "header__subtitle",
              },
              "ref": null,
              "rendered": "Perfectly Ground Work Orders",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "CREATE ORDER",
                "className": "header__button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "CREATE ORDER",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "closeModal": [Function],
            "showModal": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
